hack/release/common.sh

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+config(){
+  GITHUB_ACCOUNT="Azure"
+  RELEASE_ACR=${RELEASE_ACR:-ksnap.azurecr.io} # will always be ovreridden
+  RELEASE_REPO_ACR=${RELEASE_REPO_ACR:-${RELEASE_ACR}/public/aks/karpenter/}
+  RELEASE_REPO_MAR=mcr.microsoft.com/aks/karpenter
+  SNAPSHOT_ACR=${SNAPSHOT_ACR:-ksnap.azurecr.io}
+  SNAPSHOT_REPO_ACR=${SNAPSHOT_REPO_ACR:-${SNAPSHOT_ACR}/karpenter/snapshot/}
+
+  CURRENT_MAJOR_VERSION="0"
+  RELEASE_PLATFORM="--platform=linux/amd64,linux/arm64"
+
+  MAIN_GITHUB_ACCOUNT="Azure"
+  RELEASE_TYPE_STABLE="stable"
+  RELEASE_TYPE_SNAPSHOT="snapshot"
+}
+
+# versionData sets all the version properties for the passed release version. It sets the values
+# RELEASE_VERSION_MAJOR, RELEASE_VERSION_MINOR, and RELEASE_VERSION_PATCH to be used by other scripts
+versionData(){
+  local VERSION="$1"
+  local VERSION="${VERSION#[vV]}"
+  RELEASE_VERSION_MAJOR="${VERSION%%\.*}"
+  RELEASE_VERSION_MINOR="${VERSION#*.}"
+  RELEASE_VERSION_MINOR="${RELEASE_VERSION_MINOR%.*}"
+  RELEASE_VERSION_PATCH="${VERSION##*.}"
+  RELEASE_MINOR_VERSION="v${RELEASE_VERSION_MAJOR}.${RELEASE_VERSION_MINOR}"
+}
+
+snapshot() {
+  RELEASE_VERSION=$1
+  echo "Release Type: snapshot
+Release Version: ${RELEASE_VERSION}
+Commit: $(git rev-parse HEAD)
+Helm Chart Version $(helmChartVersion "$RELEASE_VERSION")"
+
+  authenticatePrivateRepo
+  buildImages "${SNAPSHOT_REPO_ACR}"
+  updateHelmChart
+  # not locking artifacts for snapshot releases
+  cosignImage "${CONTROLLER_IMG}"
+  cosignImage "${CONTROLLER_IMG_NAP}"
+  publishHelmChart "karpenter" "${RELEASE_VERSION}" "${SNAPSHOT_REPO_ACR}"
+  publishHelmChart "karpenter-crd" "${RELEASE_VERSION}" "${SNAPSHOT_REPO_ACR}"
+}
+
+release() {
+  RELEASE_VERSION=$1
+  echo "Release Type: stable
+Release Version: ${RELEASE_VERSION}
+Commit: $(git rev-parse HEAD)
+Helm Chart Version $(helmChartVersion "$RELEASE_VERSION")"
+
+  authenticatePrivateRepo
+  buildImages "${RELEASE_REPO_ACR}"
+  updateHelmChart "${RELEASE_REPO_MAR}"
+  lockImage "${IMG_REPOSITORY}" "${IMG_TAG}"
+  lockImage "${IMG_REPOSITORY}" "${IMG_TAG}-aks"
+  cosignImage "${CONTROLLER_IMG}"
+  cosignImage "${CONTROLLER_IMG_NAP}"
+  updateHelmChart
+  publishHelmChart "karpenter" "${RELEASE_VERSION}" "${RELEASE_REPO_ACR}"
+  publishHelmChart "karpenter-crd" "${RELEASE_VERSION}" "${RELEASE_REPO_ACR}"
+}
+
+authenticatePrivateRepo() {
+  az acr login -n "${SNAPSHOT_REPO_ACR}"
+}
+
+buildImages() {
+  RELEASE_REPO=$1
+  # Set the SOURCE_DATE_EPOCH and KO_DATA_DATE_EPOCH values for reproducable builds with timestamps
+  # https://ko.build/advanced/faq/
+
+  CONTROLLER_IMG=$(GOFLAGS=${GOFLAGS} \
+    SOURCE_DATE_EPOCH=$(git log -1 --format='%ct') KO_DATA_DATE_EPOCH=$(git log -1 --format='%ct') KO_DOCKER_REPO=${RELEASE_REPO} \
+    ko publish -B --sbom none -t "${RELEASE_VERSION}" "${RELEASE_PLATFORM}" ./cmd/controller)
+  CONTROLLER_IMG_NAP=$(GOFLAGS="${GOFLAGS} -tags=ccp" \
```

Comment:
should `ccp` be mentioned here?

Reply:
As Alex said, this part would not limited to only work with NAP, but also with having a general case of Karpenter being on a separate cluster/control plane (hence the name "ccp"). 

If it is not yet in our goals(?) to support self-hosted to be deployed like that, I am good with renaming it to "nap".

Reply:
Took another look at the code - and the alternative operator implementation has many references to ccPlaneCtx, which make a lot of sense in that context. Feel like changing the build tag would introduce too much confusion, compared to the benefits. Leaving as is.